### DISCUSSION TITLE
Spikes in response times

### DISCUSSION THREAD

**Discussion by hassanov112** [unikraft/unikraft#45] [Show and tell]
Hi, Me and a classmate did our thesis where we tested Unikraft and Docker response times and memory usage, a problem we found with Unikraft is that its response time randomly jumps up very much higher than the average response time. 

A experiment we did was to test the response time of both an Unikraft unikernel and a Docker container, as you can see in the graph below the unikernel had a better performance than the container did but the average response time was weighted down by the spikes.

Unikernel's average response time is 18,9 % short than containers average response time but it could be 29,7% shorter if the spikes is fixed. We did a light version of the same experiment to simulate a light weight program, unikernel had a longer average response time but it was shorter when we didn't count the spikes. 

![image](https://user-images.githubusercontent.com/3394601/83904946-18534c80-a761-11ea-9bb3-8ab22905f729.png)

We haven't tested why those spikes appear and therefore can not help you with it.

If you want to download the experiments and test them locally, you can find them [here](https://gitlab.com/hassanov123/unikernels-vs.-containers) (some text is in Swedish, you can use Google translate :) )

**Comment by razvand** [comment]
@hassanov112, since we are already (viciously) using [the Unikraft Discord server](https://bit.ly/UnikraftDiscord) for discussions, meetings, brainstorming and jokes (mostly bad, we do what we can), we will close the Discussions tab. We will do this on Friday, March 31, 2023.

Please move any discussions or questions on [the Unikraft Discord server](https://bit.ly/UnikraftDiscord). See you all there! :handshake:
